lib/index.ts

```diff
@@ -877,6 +884,10 @@ export function emit(rootDecl: TopLevelDeclaration, { rootFlags = ContextFlags.N
                     writeDelimited(e.members, ' | ', writeUnionReference);
                     break;
 
+                case "intersection":
+                    writeDelimited(e.members, ' & ', writeReference)

```

Comment:
This needs to be `writeUnionReference` (which can be renamed) so that we don't emit the wrong thing in cases like `() => string & number` vs `(() => string) & number`

Reply:
> There are only two hard things in Computer Science: cache invalidation and naming things.

`writeGroupedReference`, `writeEscapedReference`, `writeGuardedReference`, `writeSequencedReference`... I can't come up with a good name, might as well keep it `writeUnionReference` :)

Will update soon.

Reply:
If we want to be fancy, `writeReference` can take a `PrecedentLevel` parameter (enum) that defines what level of precedence the surrounding context is so it can intelligently add parentheses.

Reply:
That would be very nice design, and as much as I love fancy code, my just-hack-this-in-fast-and-keep-working time allowance doesn't suffice for that.